src/core/MOM_open_boundary.F90

```diff
@@ -346,6 +357,18 @@ module MOM_open_boundary
                                              !! When locked=.true.,no more boundaries can be registered.
 end type OBC_registry_type
 
+!> Type to carry OBC information needed for setting segments for OBGC tracers
+type, private :: external_tracers_segments_props
+   type(external_tracers_segments_props), pointer :: next => NULL() !< pointer to the next node
+   character(len=128) :: tracer_name      !< tracer name
+   character(len=128) :: tracer_src_file  !< tracer source file for BC
+   character(len=128) :: tracer_src_field !< name of the field in source file to extract BC
+   real               :: lfac_in  !< multiplicative factor for inbound  tracer reservoir length scale
+   real               :: lfac_out !< multiplicative factor for outbound tracer reservoir length scale
+end type external_tracers_segments_props
+!> Keeps the OBC segment properties for external tracers
```

Comment:
As described at https://github.com/NOAA-GFDL/MOM6/wiki/Code-style-guide#global--module-data, MOM6 does not permit the use of module data, apart from CPU clocks, apart from a few specific exceptions that do not apply to the new variables being introduced here.  Please refactor this contribution in alignment with this policy.